mk/docker.mk

```diff
@@ -1,197 +1,132 @@
+	docker build -t $$(call build_image,kuma-dp,$(1)) --build-arg ARCH=$(1) --platform=linux/$(1) -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.kuma-dp .
 
-.PHONY: image/kuma-dp
-image/kuma-dp: image/base image/envoy build/kuma-dp/linux-${GOARCH} build/coredns/linux-${GOARCH} build/artifacts-linux-${GOARCH}/envoy/envoy ## Dev: Rebuild `kuma-dp` Docker image
-	docker build -t $(KUMA_DP_DOCKER_IMAGE) ${DOCKER_BUILD_ARGS} --build-arg ARCH=${GOARCH} --platform=linux/${GOARCH} -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.kuma-dp .
+.PHONY: image/kumactl/$(1)
+image/kumactl/$(1): image/base/$(1) build/artifacts-linux-$(1)/kumactl ## Dev: Rebuild `kumactl` Docker image
+	docker build -t $$(call build_image,kumactl,$(1)) --build-arg ARCH=$(1) --platform=linux/$(1) -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.kumactl .
 
-.PHONY: image/kumactl
-image/kumactl: image/base build/kumactl/linux-${GOARCH} ## Dev: Rebuild `kumactl` Docker image
-	docker build -t $(KUMACTL_DOCKER_IMAGE) ${DOCKER_BUILD_ARGS} --build-arg ARCH=${GOARCH} --platform=linux/${GOARCH} -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.kumactl .
+.PHONY: image/kuma-init/$(1)
+image/kuma-init/$(1): build/artifacts-linux-$(1)/kumactl ## Dev: Rebuild `kuma-init` Docker image
+	docker build -t $$(call build_image,kuma-init,$(1)) --build-arg ARCH=$(1) --platform=linux/$(1) -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.kuma-init .
 
-.PHONY: image/kuma-init
-image/kuma-init: build/kumactl/linux-${GOARCH} ## Dev: Rebuild `kuma-init` Docker image
-	docker build -t $(KUMA_INIT_DOCKER_IMAGE) ${DOCKER_BUILD_ARGS} --build-arg ARCH=${GOARCH} --platform=linux/${GOARCH} -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.kuma-init .
+.PHONY: image/kuma-cni/$(1)
+image/kuma-cni/$(1): image/base-root/$(1) build/artifacts-linux-$(1)/kuma-cni build/artifacts-linux-$(1)/install-cni
+	docker build -t $$(call build_image,kuma-cni,$(1)) --build-arg ARCH=$(1) --platform=linux/$(1) -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.kuma-cni .
 
-.PHONY: image/kuma-cni
-image/kuma-cni: image/base-root build/kuma-cni/linux-${GOARCH} build/install-cni/linux-${GOARCH}
-	docker build -t $(KUMA_CNI_DOCKER_IMAGE) ${DOCKER_BUILD_ARGS} --build-arg ARCH=${GOARCH} --platform=linux/${GOARCH} -f $(TOOLS_DIR)/releases/dockerfiles/Dockerfile.kuma-cni .
+.PHONY: image/kuma-universal/$(1)
+image/kuma-universal/$(1): image/envoy/$(1) build/artifacts-linux-$(1)/kuma-cp build/artifacts-linux-$(1)/kuma-dp build/artifacts-linux-$(1)/kumactl build/artifacts-linux-$(1)/kumactl build/artifacts-linux-$(1)/test-server
+	docker build -t $$(call build_image,kuma-universal,$(1)) --build-arg ARCH=$(1) --platform=linux/$(1) -f $(KUMA_DIR)/test/dockerfiles/Dockerfile.universal .
+endef
+$(foreach goarch,$(SUPPORTED_GOARCHES),$(eval $(call IMAGE_TARGETS_BY_ARCH,$(goarch))))
+
+# add targets to generate docker/{save,load,tag,push} for each supported ARCH
+define DOCKER_TARGETS_BY_ARCH
```

Comment:
what do you think about adopting the pattern of documenting args for define functions? For example
```
# $(1) - docker image
# $(2) - architecture
define DOCKER_TARGETS_BY_ARCH
```
Because they are not named, I have to go and see the usage of this function to figure out what this is.